bob-access/src/authenticator/basic.rs

```diff
@@ -60,8 +62,22 @@ impl<Storage: UsersStorage> Basic<Storage> {
                 );
         
                 let user = self.users_storage.get_user(&username)?;
-                if user.password() == password {
-                    Ok(user.into())
+                if let Some(usr_password) = user.password() {
+                    if usr_password == password {
+                        Ok(user.into())
+                    } else {
+                        Err(Error::UnauthorizedRequest)
+                    }
+                } else if let Some(usr_hash) = user.hash() {
+                    let hash_str = format!("{}bob", password);
```

Comment:
I think, we can make salt configurable by adding optional root parameter to `users.yaml`